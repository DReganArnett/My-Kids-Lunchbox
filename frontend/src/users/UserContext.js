import React from 'react';

// Provides user object and allows for setting

const UserContext = React.createContext();

export default UserContext;