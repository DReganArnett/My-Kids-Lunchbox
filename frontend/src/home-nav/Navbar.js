import React, {useContext} from 'react';
import {Link, NavLink} from 'react-router-dom';
import UserContext from '../users/UserContext';

const Navbar = ({logout}) => {
    const {currentUser} = useContext(UserContext);

    const navIfLoggedIn = () => {
        return (
            <div className="Navbar-loggedIn">
                <span><button><NavLink to="/">Home</NavLink></button></span>
                <span><button><NavLink to="/lunches">Lunches</NavLink></button></span>
                <span><button><NavLink to="/createlunch">My Kitchen</NavLink></button></span>
                <span><button><NavLink to="/users/reviews">My Reviews</NavLink></button></span>
                <span><button><NavLink to="/users/favorites">My Favorites</NavLink></button></span>
                <span><button><NavLink to="/users/profile/">My Profile</NavLink></button></span>
                <span><button><Link to="/" onClick={logout}>Log Out</Link></button></span>
            </div> 
        );
    }

    const navIfLoggedOut = () => {
        return (
            <div className='Navbar-loggedOut'>
                <span><button><NavLink to="/">Home</NavLink></button></span>
                <span><button><NavLink to="/login">Log In</NavLink></button></span>
                <span><button><NavLink to="/register">Sign Up</NavLink></button></span>
            </div>
        );
    }

    return (
        
        <div className='Navbar'>
            {currentUser ? navIfLoggedIn() : navIfLoggedOut()}
        </div>
    );
}

export default Navbar;