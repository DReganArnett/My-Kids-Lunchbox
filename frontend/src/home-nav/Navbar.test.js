import React from 'react';
import { BrowserRouter as Router} from 'react-router-dom';
import {render, screen} from '@testing-library/react';
import UserContext from '../users/UserContext';
import Navbar from '../home-nav/Navbar';


jest.mock('../users/UserContext');

// Smoke test
it('should render without crashing', function() {
  const mockContext = {
    currentUser: {username: "testuser1"},
    logout: jest.fn()
  };  

  UserContext._currentValue = mockContext;

  render(
    <Router>
      <Navbar />
    </Router>
  );
});

// Snapshot test
it('should match snapshot', () => {
  const mockContext = {
    currentUser: {username: "testuser1"},
    logout: jest.fn()
  };

  UserContext._currentValue = mockContext;

  const {asFragment} = render(
    <Router>
      <Navbar />
    </Router>
  );

  expect(asFragment()).toMatchSnapshot();
});

