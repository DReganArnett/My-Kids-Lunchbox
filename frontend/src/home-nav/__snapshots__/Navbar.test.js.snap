// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Navbar"
  >
    <div
      class="Navbar-loggedIn"
    >
      <span>
        <button>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </button>
      </span>
      <span>
        <button>
          <a
            href="/lunches"
          >
            Lunches
          </a>
        </button>
      </span>
      <span>
        <button>
          <a
            href="/createlunch"
          >
            My Kitchen
          </a>
        </button>
      </span>
      <span>
        <button>
          <a
            href="/users/reviews"
          >
            My Reviews
          </a>
        </button>
      </span>
      <span>
        <button>
          <a
            href="/users/favorites"
          >
            My Favorites
          </a>
        </button>
      </span>
      <span>
        <button>
          <a
            href="/users/profile/"
          >
            My Profile
          </a>
        </button>
      </span>
      <span>
        <button>
          <a
            href="/"
          >
            Log Out
          </a>
        </button>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
